### ISSUE TITLE
OpenTelemetry - respect proxy settings in VertxGrpcExporter and VertxHttpExporter

### ISSUE THREAD

**Issue by michal-pasinski-db** [quarkusio/quarkus#39519]
### Description

Starting from quarkus version `3.3.x` the spans are exported using Vert.x based exporters. It is not possible to set Proxy on the Vert.x Http client used by these exporters (analog to `quarkus.rest-client.my-client.proxy-address`). 

Expected: Via `quarkus.otel.exporter.otlp.traces.proxy-address` it should be possible to configure a proxy server to be used to communicate with external server.

### Implementation ideas

1. Extend `OtlpExporterTracesConfig` with `proxyAddress` property.
2. Extend `io.quarkus.opentelemetry.runtime.exporter.otlp.OTelExporterRecorder.HttpClientOptionsConsumer#HttpClientOptionsConsumer` with `proxyAddress` and add private method `configureProxy`

With the above proposed small changes, it will be possible to use these exporters behind a proxy

_No response_

**Comment by geoand** [comment]
Makes sense, I'll look into it!

**Comment by geoand** [comment]
Question, prior to 3.3 how did you configure these settings?

**Comment by geoand** [comment]
In any case I opened #39543 which uses the same approach as we have in the REST Client

**Comment by michal-pasinski-db** [comment]
@geoand Prior to 3.3 we did not configure it explicitly. I presume that OkHttpClient was happy with our `HTTP_PROXY` and `HTTPS_PROXY` environment variables. The update to 3.3 resulted in our spans not being sent as Vert.x client could not reach our opentelemetry server. We did some tweaking on our own to set this proxy exaclty the way you set it in `configureProxyOptions` and our spans gets sent correcly.

Thank you very much for such a quick reaction!
